sdcm/logcollector.py

```diff
@@ -1134,6 +1136,34 @@ def is_collect_to_a_single_archive(self) -> bool:
         return True
 
 
+class SSTablesCollector(SCTLogCollector):
+    """
+    Collect corrupted sstables from db nodes.
+    """
+    log_entities = [
+        FileLog(name='parallel-timelines-report.html',
+                search_locally=True)
+    ]
+    cluster_log_type = "corrupted-sstables"
+    cluster_dir_prefix = "corrupted-sstables"
+
+    def collect_logs(self, local_search_path: Optional[str] = None) -> list[str]:
+        s3_links = []
+        raw_events_file_path = Path(self.local_dir).parent.parent.parent / EVENTS_LOG_DIR / RAW_EVENTS_LOG
+        with open(raw_events_file_path, "r", encoding="utf-8") as events_file:
+            node_names = {event.get("node") for raw_line in events_file.readlines()
+                          if (event := json.loads(raw_line)).get("type") == "MALFORMED_SSTABLE"}
+        nodes_with_malformed_sstable: List[CollectingNode] = [node for node in self.nodes if node.name in node_names]
+        for node in nodes_with_malformed_sstable:
+            LOGGER.info("Collecting sstables for node %s...", node.name)
+            node.remoter.sudo("systemctl stop scylla-server.service")
```

Comment:
Why are we stopping Scylla ? sstables written to disk shouldn't be modified again.

Are we copying the whole table ? not the specific sstables marked identify as courrupted ?

Reply:
IIUC this is to avoid this sstable will be compacted, and will be deleted

Reply:
so doing that at the log collection step might be a bit too late, I would argue that the whole process need to be done as close to finding it in the logs as possible.

Reply:
this will now raise a critical event, so i'm not sure that it will be a big problem

Reply:
If we copy only corrupted files, then there's no need to stop scylla. Otherwise we don't want to compress/send files while they are changed (e.g. new sstables, compactions).

Reply:
I removed stopping scylla - instead create snapshot and take sstables from it.